src/Drupal/Commands/core/LocaleCommands.php

```diff
@@ -336,4 +398,76 @@ private function isTranslatable(LanguageInterface $language)
             ->get('locale.settings')
             ->get('translate_english');
     }
+
+    /**
+     * Get PODatabaseReader options for given types.
+     *
+     * @param array $types
+     * @return array
+     *   Options list with value 'true'.
+     * @throws \Exception
+     *   Triggered with incorrect types.
+     */
+    private function convertTypesToPoDbReaderOptions(array $types = [])
+    {
+        $valid_convertions = [
+            'not_customized' => 'not-customized',
+            'customized' => 'customized',
+            'not_translated' => 'not-translated',
+        ];
+
+        if (empty($types)) {
+            return array_fill_keys(array_keys($valid_convertions), true);
+        }
+
+        // Check for invalid conversions.
+        if (array_diff($types, $valid_convertions)) {
+            throw new \Exception(dt('Allowed types: @types.', [
+                '@types' => implode(', ', $valid_convertions),
+            ]));
+        }
+
+        // Convert Types to Options.
+        $options = array_keys(array_intersect($valid_convertions, $types));
+
+        return array_fill_keys($options, true);
+    }
+
+    /**
+     * Write out the exported language or template file.
+     *
+     * @param string $file_uri Uri string to gather the data.
+     * @param LanguageInterface|null $language The language to export.
+     * @param array $options The export options for PoDatabaseReader.
+     * @return bool True if successful.
+     */
+    private function writePoFile($file_uri, LanguageInterface $language = null, array $options = [])
+    {
+        $reader = new PoDatabaseReader();
+
+        if ($language) {
+            $reader->setLangcode($language->getId());
+            $reader->setOptions($options);
+        }
+
+        $reader_item = $reader->readItem();
+        if (empty($reader_item)) {
+            return false;
+        }
+
+        $header = $reader->getHeader();
+        $header->setProjectName(drush_get_context('DRUSH_DRUPAL_SITE'));
```

Comment:
We are trying to get rid of calls to drush_get_context(). Can you do something like Drupal::config()->get('system.site_name')? Or Drush::uri()

I'll try to more of a review soon.